TopMovies/Core/State/MoviesListState.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  MoviesListState.swift
+//  TopMovies
+//
+//  Created by anikolaenko on 02.02.2021.
+//
+
+import ReSwift
+
+enum DataState<T> {
+    case initial
+    case loading
+    case data(T)
+    case noData
+}
```

Comment:
Good plan, but a bit uncompleted. I'd prefer `completed(data: T)` case instead of `data` and `noData`. 
Moreover, there shouldn't be optional case here. If there's no data, there has to be an error - `case .failed(Error)`.

Reply:
And let's call it `RequestState` and move it in a separate file to reuse in other states next time.